llvm/utils/lit/lit/TestRunner.py

```diff
@@ -87,10 +87,11 @@ class ShellEnvironment(object):
     we maintain a dir stack for pushd/popd.
     """
 
-    def __init__(self, cwd, env):
+    def __init__(self, cwd, env, umask=-1):
         self.cwd = cwd
         self.env = dict(env)
         self.dirStack = []
+        self.umask = umask
```

Comment:
This is just a nit but I'd reorder these two fields to match the order of arguments.

```suggestion
        self.umask = umask
        self.dirStack = []
```

Reply:
Done, thanks.